src/rules/noObjectLiteralTypeAssertionRule.ts

```diff
@@ -37,23 +45,35 @@ export class Rule extends Lint.Rules.AbstractRule {
             The compiler will warn for excess properties with this syntax, but not missing required fields.
             For example: \`const x: { foo: number } = {}\` will fail to compile, but
             \`const x = {} as { foo: number }\` will succeed.`,
-        optionsDescription: "Not configurable.",
-        options: null,
-        optionExamples: [true],
+        optionsDescription: Lint.Utils.dedent`
+            One argument may be optionally provided:
+
+            * \`${OPTION_ALLOW_ARGUMENTS}\` allows type assertions to be used on object literals inside call expressions.`,
```

Comment:
Although TSLint has traditionally had rule arguments provided as strings, that's starting to be a bit of a pain when they become more complex. Let's have the rule take in an object:

```ts
[true, {
    "allow-arguments": true
}]
```

Reply:
Done 👍 I borrowed the new schema from the `noImportSideEffectRule.ts` rule.

https://github.com/palantir/tslint/blob/9785477f26c1e0f926a5a34ba9170f1784a728c7/src/rules/noImportSideEffectRule.ts#L29-L41